File: src/types/agent-run.ts
export type AgentRunStatus = 'running' | 'succeeded' | 'failed' | 'stopped' | 'abandoned';
export type AgentRunStreamType = 'stdout' | 'stderr' | 'system';
export type AgentRunType = 'codex' | 'claude_code';

export interface WorkspaceRun {
  id: string;
  workspaceId: string;
  agentType: AgentRunType | string;
  command: string;
  args: string[];
  cwd: string;
  status: AgentRunStatus | string;
  pid?: number;
  startedAt: string;
  finishedAt?: string;
  exitCode?: number;
  errorMessage?: string;
}

export interface WorkspaceRunLog {
  id: string;
  runId: string;
  timestamp: string;
  streamType: AgentRunStreamType | string;
  message: string;
}

export interface StartWorkspaceRunInput {
  workspaceId: string;
  agentType: AgentRunType;
  prompt?: string;
}

File: frontend/src-tauri/src/models/agent_run.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceRun {
    pub id: String,
    pub workspace_id: String,
    pub agent_type: String,
    pub command: String,
    pub args: Vec<String>,
    pub cwd: String,
    pub status: String,
    pub pid: Option<u32>,
    pub started_at: String,
    pub finished_at: Option<String>,
    pub exit_code: Option<i32>,
    pub error_message: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct WorkspaceRunLog {
    pub id: String,
    pub run_id: String,
    pub timestamp: String,
    pub stream_type: String,
    pub message: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct StartWorkspaceRunInput {
    pub workspace_id: String,
    pub agent_type: String,
    pub prompt: Option<String>,
}
